packages/loopring_v3/contracts/impl/libtransactions/AccountTransferTransaction.sol

```diff
@@ -153,6 +156,14 @@ library OwnerChangeTransaction
                     )
                 )
             );
+            // Verify that the data was signed by the current owner
```

Comment:
In the case of using stateless wallet, the current owner may have lost access, so no need to check authorization from the current owner.

Reply:
The wallet data would be signed before it is set on the account while the owner still has access to his keys. So if the owner wants to set the data on his account, he would also sign it and store the signature for it.

I added this for extra security when the user's EdDSA keys are lost or shared between different users. Currently it is allowed to update the EdDSA keys/wallet data for an account with just the EdDSA keys because it's significantly cheaper than needing an ECDSA signature. If the users's EdDSA keys are stolen all funds in his account are at risk, but when the wallet can also be updated he would also risk losing his account completely because the owner could be changed by setting some dummy wallet. By also requiring an ECDSA signature for the wallet data this is not possible. At least the user's account is safe from being overtaken completely, so the user's accountID is still secure as long as his Ethereum key is safe.

So not too important, but allows us to keep more common operations as cheap as possible while having some extra safe guards against theft of the EdDSA keys.

Reply:
"The wallet data would be signed before it is set on the account while the owner still has access to his keys"
Unless the signature of the current owner is saved inside the Merkle tree and becomes part of DA, there is no guarantee that his/her signature can be found when the user's wallet is used.

Reply:
The same is true for the actual wallet data, which is also not part of the Merkle tree or the DA, but needs to be stored somewhere else by the user/platform (publicly or privately), so no difference there.